### PR TITLE
Fix related assemblies count and Thoas gRPC bug

### PR DESCRIPTION
**Pull request by bilalebi**
### JIRA Ticket
https://www.ebi.ac.uk/panda/jira/browse/EA-1131

### Changes
* Fixed `IndexError: list index out of range` when returning related assembly count
   * I mistakingly used `fetch_organisms_group_counts()` 
   * Added a `fetch_related_assemblies_count()` which does the count for a given `species_taxonomy_id` 
* Fixed a bug where GraphQL gRPC request was able to fetch unreleased data ([EA-1131](https://www.ebi.ac.uk/panda/jira/browse/EA-1131))

### CODE REVIEW

File: src/ensembl/production/metadata/grpc/adaptors/genome.py [hunk 1]
```diff
@@ -647,3 +650,37 @@ def fetch_organisms_group_counts(self, species_taxonomy_id=None, release_version
 		with self.metadata_db.session_scope() as session:
 			# TODO check if we should return a dictionary instead
 			return session.execute(query).all()
+
+	def fetch_related_assemblies_count(self, species_taxonomy_id=None, release_version=None):
+		o_species = aliased(Organism)
+		o = aliased(Organism)
+		if not release_version:
+			# Get latest released organisms
+			query = db.select(
```
Comment (marcoooo) [inline src/ensembl/production/metadata/grpc/adaptors/genome.py:659]: Why not just counting and return the number directly here?

File: src/ensembl/production/metadata/grpc/utils.py [hunk 2]
```diff
@@ -93,9 +93,13 @@ def create_genome_with_attributes_and_count(db_conn, genome, release_version):
...
     # fetch related assemblies count
-    related_assemblies_count = db_conn.fetch_organisms_group_counts(
+    count_result = db_conn.fetch_related_assemblies_count(
         species_taxonomy_id=genome.Organism.species_taxonomy_id
-    )[0].count
+    )
+
+    related_assemblies_count = 0
+    if len(count_result) > 0:
+        related_assemblies_count = count_result[0].count
```
Comment (marcoooo) [inline src/ensembl/production/metadata/grpc/utils.py:102]: Could be simplified by 
`related_assemblies_count = count_result[0].count if len(count_result) > 0 else 0`
Or from previous comment, the method directly return a count, as its name suggests.

File: src/ensembl/production/metadata/grpc/adaptors/genome.py (hunk 1 above)
Comment (bilalebi) [reply src/ensembl/production/metadata/grpc/adaptors/genome.py:659]: That's a good idea, thank you for your suggestion

File: src/ensembl/production/metadata/grpc/utils.py (hunk 2 above)
Comment (bilalebi) [reply src/ensembl/production/metadata/grpc/utils.py]: I implemented your suggestion above

File: src/ensembl/production/metadata/grpc/adaptors/genome.py (hunk 1 above)
Comment (bilalebi) [reply src/ensembl/production/metadata/grpc/adaptors/genome.py:659]: Done

File: src/ensembl/production/metadata/grpc/adaptors/genome.py [hunk 3]
```diff
@@ -647,3 +650,39 @@ def fetch_organisms_group_counts(self, species_taxonomy_id=None, release_version
...
+				o_species.scientific_name,
+			)
+		else:
+			# change group to release_version_state and related genomes
+			raise NotImplementedError('Not implemented yet')
+			pass
+
+		# print(f"query ---> {query}")
+		with self.metadata_db.session_scope() as session:
+			results = session.execute(query).all()
```
Comment (marcoooo) [inline src/ensembl/production/metadata/grpc/adaptors/genome.py:685]: Was more thinking of something like: 

`session.execute(query).count()` 

will return all you need automatically, no need to reinvent the weel, so you don't do the `group` in the query :-)

File: src/ensembl/production/metadata/grpc/adaptors/genome.py [hunk 4]
```diff
@@ -647,3 +650,39 @@ def fetch_organisms_group_counts(self, species_taxonomy_id=None, release_version
...
+				o_species.scientific_name,
+				db.func.count().label('count')
+			)
+
+			query = query.join(o, o_species.species_taxonomy_id == o.species_taxonomy_id)
+			query = query.join(Genome, o.organism_id == Genome.organism_id)
+			query = query.join(Assembly, Genome.assembly_id == Assembly.assembly_id)
+			query = query.filter(o_species.species_taxonomy_id == species_taxonomy_id)
+
+			query = query.group_by(
```
Comment (marcoooo) [inline src/ensembl/production/metadata/grpc/adaptors/genome.py:672]: No need to group or else.

File: src/ensembl/production/metadata/grpc/utils.py (hunk 2 above)
Comment (marcoooo) [reply src/ensembl/production/metadata/grpc/utils.py]: I just did :-)

**Comment by marcoooo** [comment]
> ### JIRA Ticket
> 
> https://www.ebi.ac.uk/panda/jira/browse/EA-1131
> ### Changes
> 
>     * Fixed `IndexError: list index out of range` when returning related assembly count
>       
>       * I mistakingly used `fetch_organisms_group_counts()`
>       * Added a `fetch_related_assemblies_count()` which does the count for a given `organisms_id`
> 
>     * Fixed a bug where GraphQL gRPC request was able to fetch unreleased data ([EA-1131](https://www.ebi.ac.uk/panda/jira/browse/EA-1131))

**Review by dpopleton — APPROVED** [review]

**Review by kamaldodiya — APPROVED** [review]

**Comment by kamaldodiya** [comment]
Response in web which was broken earlier 
```
curl 'http://localhost:8014/api/metadata/genome/d1d26e8d-cd86-4c69-b2d3-13bfd702b033/explain' | jq
{
  "genome_id": "d1d26e8d-cd86-4c69-b2d3-13bfd702b033",
  "genome_tag": null,
  "species_taxonomy_id": "10089",
  "common_name": "Ryukyu mouse",
  "scientific_name": "Mus caroli",
  "type": {
    "kind": "strain",
    "value": null
  },
  "is_reference": false,
  "assembly": {
    "accession_id": "GCA_900094665.2",
    "name": "CAROLI_EIJ_v1.1"
  }
}
```

**Pull request merged**